lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -67,28 +68,113 @@ class ScfWhileOp : public calyx::WhileOpInterface<scf::WhileOp> {
   std::optional<uint64_t> getBound() override { return std::nullopt; }
 };
 
+class ScfForOp : public calyx::RepeatOpInterface<scf::ForOp> {
+public:
+  explicit ScfForOp(scf::ForOp op) : calyx::RepeatOpInterface<scf::ForOp>(op) {}
+
+  Block::BlockArgListType getBodyArgs() override {
+    return getOperation().getLoopBody().getArguments();
+  }
+
+  Block *getBodyBlock() override {
+    return &getOperation().getLoopBody().getBlocks().front();
+  }
+
+  std::optional<uint64_t> getBound() override {
+    return constantTripCount(getOperation().getLowerBound(),
```

Comment:
This returns an int64_t. Can we always make getBound() return an int64_t?

Reply:
The type for `getBound()` comes from `BasicLoopInterface` and is implemented in many contexts, not just for loops. At least to me, it seems like (since we don't always know the bound) it makes intuitive sense to return `std::optional`? lmk if I'm missing something, though

Reply:
Ah yeah I meant keep std::optional, but return std::optional<int64_t> instead of std::optional<uint64_t>. However, what you say is right; this is overridden. I missed that. Sorry!

Reply:
I see, I didn't realize `constantTripCount` returns an `int64_t`... seems a bit weird, since intuitively, how would the trip count ever be negative? Either way, I'll write up an issue.

Reply:
Integers usually lend to safer code; perhaps this is an API choice to avoid an accidental `0 - 1` which would result in a very large trip count, and thus something like `for (int i = 0; i < *getBound(); ++i) { ... }` would run for a very long time.